gitlabform/gitlab/resource_group.py

```diff
@@ -0,0 +1,22 @@
+from gitlabform.gitlab.projects import GitLabProjects
+
+
+class GitLabResourceGroup(GitLabProjects):
+    def get_specific_resource_group(self, project_and_group_name, resource_group_name):
+        return self._make_requests_to_api(
+            "projects/%s/resource_groups/%s",
+            (project_and_group_name, resource_group_name),
+            method="GET",
+            expected_codes=[200, 404],
```

Comment:
Why are you accepting 404 status code here? I am not saying that it has to be wrong, but we should have a good reason for it. In other places in the code we only do that for the _delete_ methods, for idempotency.

Reply:
There's no reason other than putting all the possible status codes from the API, which doesn't make sense now that I'm looking at it.

Reply:
Actually never mind, after doing some local testing, I have more control over the error handling by placing the 400 codes in the expected codes. 

e.g.

```
(venv) EP-ML-10467:gitlabform linden.kyaw$ gitlabform --config ep-my-project.yaml --output-file ep-my-project-output.yaml ALL_DEFINED
🏗  GitLabForm version: 3.0.0rc2 = outdated 😔, please update! (the latest stable is 3.0.0.post1)
>>> Getting ALL_DEFINED groups and projects...
:: # of groups to process: 0
:: # of projects to process: 1
* (1/1) Processing project: linden.kyaw/my.project
Warning: Error occurred while processing project linden.kyaw/my.project, exception:

Resource with url='https://gitlab.elasticpath.com//api/v4/projects/linden.kyaw%2Fmy.project/resource_groups/abs?per_page=100' not found (HTTP 404)!
:: # of groups processed successfully: 0
:: # of projects processed successfully: 0
:: # of projects failed: 1 
:: Failed project 1: linden.kyaw/my.project 
(venv) EP-ML-10467:gitlabform linden.kyaw$ gitlabform --config ep-my-project.yaml --output-file ep-my-project-output.yaml ALL_DEFINED
🏗  GitLabForm version: 3.0.0rc2 = outdated 😔, please update! (the latest stable is 3.0.0.post1)
>>> Getting ALL_DEFINED groups and projects...
:: # of groups to process: 0
:: # of projects to process: 1
* (1/1) Processing project: linden.kyaw/my.project
Error: Project is not configured to use resource group: abs
```

Reply:
404 Removed.